libs/compute_cuda/CMakeLists.txt

```diff
@@ -40,12 +43,19 @@ include(HPX_AddModule)
 add_hpx_module(compute_cuda
   FORCE_LINKING_GEN
   GLOBAL_HEADER_GEN OFF
+  EXCLUDE_FROM_GLOBAL_HEADER
+    hpx/compute/cuda/detail/launch.hpp
+    hpx/compute/cuda/detail/scoped_active_target.hpp
+    hpx/compute/detail/get_proxy_type.hpp
+    hpx/compute/detail/iterator.hpp
+    hpx/compute/detail/target_distribution_policy.hpp
```

Comment:
I think instead of doing this manually we can just add a check to exclude any headers with `/detail/` in their path over here: https://github.com/STEllAR-GROUP/hpx/blob/502eeb4f8c03fab421756f3a63f69760cdd969d4/cmake/HPX_AddModule.cmake#L98-L100.

Reply:
I thought about that but some of the detail headers were needed for example `execution_information` using `rp.get_pu_mask()` resulted in an `access to incomplete type 'hpx::resource::detail::partitioner' error`
I guessed I make some little changes in the exposed `partitioner.hpp` and exclude detail/ in the `HPX_AddModule.cmake`
Will do

Reply:
We can explicitly include a detail header in cases like that.

Reply:
Yes, that only tells us that we didn't #include the `/detail` header in places where it's needed. Let's fix that instead.